custom_metrics/origin-trials.js

```diff
@@ -0,0 +1,156 @@
+//[origin_trials]
+/**
+ * Origin Trials
+ * https://github.com/GoogleChrome/OriginTrials
+ *
+ * Test site: https://floc-ot-meta.glitch.me/
+ * Test site: https://floc-ot-header.glitch.me/
+ */
+
+/**
+ * https://github.com/GoogleChrome/OriginTrials/blob/gh-pages/check-token.html
+ *
+ * @function validate
+ * @param {string} tokenElem
+ * @returns {object} origin_trial_metadata
+ */
+function validate(tokenElem) {
+  let validityElem,
+    versionElem,
+    originElem,
+    subdomainElem,
+    thirdpartyElem,
+    usageElem,
+    featureElem,
+    expiryElem;
+
+  const utf8Decoder = new TextDecoder('utf-8', {fatal: true});
+
+  // Base64-decode the token into a Uint8Array.
+  let tokenStr;
+  try {
+    tokenStr = atob(tokenElem);
+  } catch (e) {
+    console.error(e);
+    validityElem = 'Invalid Base64';
+    return;
+  }
+  const token = new Uint8Array(tokenStr.length);
+  for (let i = 0; i < token.length; i++) {
+    token[i] = tokenStr.charCodeAt(i);
+  }
+
+  // Check that the version number is 2 or 3.
+  const version = token[0];
+  versionElem = '' + version;
+  if (version !== 2 && version !== 3) {
+    validityElem = 'Unknown version';
+    return;
+  }
+
+  // Pull the fields out of the token.
+  if (token.length < 69) {
+    validityElem = 'Token is too short';
+    return;
+  }
+  const payloadLength = new DataView(token.buffer, 65, 4).getInt32(0, /*littleEndian=*/ false);
+  const payload = new Uint8Array(token.buffer, 69);
+  if (payload.length !== payloadLength) {
+    validityElem = 'Token is ' + payload.length + ' bytes; expected ' + payloadLength;
+    return;
+  }
+
+  // The version + length + payload is signed.
+  const signedData = new Uint8Array(token.buffer.slice(64));
+  signedData[0] = token[0];
+
+  // Pull the fields out of the JSON payload.
+  let json;
+  try {
+    json = utf8Decoder.decode(payload);
+  } catch (e) {
+    console.error(e);
+    validityElem = 'Invalid UTF-8';
+    return;
+  }
+
+  let obj;
+  try {
+    obj = JSON.parse(json);
+  } catch (e) {
+    console.error(e);
+    validityElem = 'Invalid JSON';
+    return;
+  }
+
+  originElem = obj.origin;
+  subdomainElem = obj.isSubdomain ? 'Yes' : 'No';
+  thirdpartyElem = obj.isThirdParty ? 'Yes' : 'No';
```

Comment:
Seems like it'd be more useful to leave these as booleans, rather than having to do a string comparison later during aggregation/analysis.
```suggestion
  subdomainElem = !!obj.isSubdomain;
  thirdpartyElem = !!obj.isThirdParty;
```